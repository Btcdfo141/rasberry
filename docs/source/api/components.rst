:mod:`spencerassistant.components`
===============================

air\_quality
--------------------------------------------

.. automodule:: spencerassistant.components.air_quality
    :members:
    :undoc-members:
    :show-inheritance:

alarm\_control\_panel
--------------------------------------------

.. automodule:: spencerassistant.components.alarm_control_panel
    :members:
    :undoc-members:
    :show-inheritance:

binary\_sensor
--------------------------------------------

.. automodule:: spencerassistant.components.binary_sensor
    :members:
    :undoc-members:
    :show-inheritance:

camera
---------------------------

.. automodule:: spencerassistant.components.camera
    :members:
    :undoc-members:
    :show-inheritance:

calendar
---------------------------

.. automodule:: spencerassistant.components.calendar
    :members:
    :undoc-members:
    :show-inheritance:

climate
---------------------------

.. automodule:: spencerassistant.components.climate
    :members:
    :undoc-members:
    :show-inheritance:

conversation
---------------------------

.. automodule:: spencerassistant.components.conversation
    :members:
    :undoc-members:
    :show-inheritance:

cover
---------------------------

.. automodule:: spencerassistant.components.cover
    :members:
    :undoc-members:
    :show-inheritance:

device\_tracker
---------------------------

.. automodule:: spencerassistant.components.device_tracker
    :members:
    :undoc-members:
    :show-inheritance:

fan
---------------------------

.. automodule:: spencerassistant.components.fan
    :members:
    :undoc-members:
    :show-inheritance:

light
---------------------------

.. automodule:: spencerassistant.components.light
    :members:
    :undoc-members:
    :show-inheritance:

lock
---------------------------

.. automodule:: spencerassistant.components.lock
    :members:
    :undoc-members:
    :show-inheritance:

media\_player
---------------------------

.. automodule:: spencerassistant.components.media_player
    :members:
    :undoc-members:
    :show-inheritance:

notify
---------------------------

.. automodule:: spencerassistant.components.notify
    :members:
    :undoc-members:
    :show-inheritance:

remote
---------------------------

.. automodule:: spencerassistant.components.remote
    :members:
    :undoc-members:
    :show-inheritance:

switch
---------------------------

.. automodule:: spencerassistant.components.switch
    :members:
    :undoc-members:
    :show-inheritance:

sensor
-------------------------------------

.. automodule:: spencerassistant.components.sensor
    :members:
    :undoc-members:
    :show-inheritance:

vacuum
-------------------------------------

.. automodule:: spencerassistant.components.vacuum
    :members:
    :undoc-members:
    :show-inheritance:

water\_heater
-------------------------------------

.. automodule:: spencerassistant.components.water_heater
    :members:
    :undoc-members:
    :show-inheritance:

weather
---------------------------

.. automodule:: spencerassistant.components.weather
    :members:
    :undoc-members:
    :show-inheritance:

webhook
---------------------------

.. automodule:: spencerassistant.components.webhook
    :members:
    :undoc-members:
    :show-inheritance:
