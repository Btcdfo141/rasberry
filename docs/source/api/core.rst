.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: