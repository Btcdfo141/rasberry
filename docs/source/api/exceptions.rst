.. _exceptions_module:

:mod:`spencerassistant.exceptions`
-------------------------------

.. automodule:: spencerassistant.exceptions
    :members:
