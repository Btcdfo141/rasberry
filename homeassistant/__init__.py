"""Init file for Home Assistant."""


test2
