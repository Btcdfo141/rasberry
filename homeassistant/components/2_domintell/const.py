"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
