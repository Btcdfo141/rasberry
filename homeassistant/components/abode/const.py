"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
