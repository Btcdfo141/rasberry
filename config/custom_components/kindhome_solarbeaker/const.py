from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
