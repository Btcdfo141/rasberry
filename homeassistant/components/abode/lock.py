"""Support for Abode Security System locks."""
import logging

from homeassistant.components.lock import LockDevice

from . import AbodeDevice
from .const import DOMAIN

_LOGGER = logging.getLogger(__name__)


async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
    """This platform uses config entries."""
    pass


async def async_setup_entry(hass, config_entry, async_add_devices):
    """Set up Abode lock devices."""
    import abodepy.helpers.constants as CONST

    data = hass.data[DOMAIN]

    devices = []
    for device in data.abode.get_devices(generic_type=CONST.TYPE_LOCK):
        if data.is_excluded(device):
            continue

        devices.append(AbodeLock(data, device))

    async_add_devices(devices)


class AbodeLock(AbodeDevice, LockDevice):
    """Representation of an Abode lock."""

    def lock(self, **kwargs):
        """Lock the device."""
        self._device.lock()

    def unlock(self, **kwargs):
        """Unlock the device."""
        self._device.unlock()

    @property
    def is_locked(self):
        """Return true if device is on."""
        return self._device.is_locked
