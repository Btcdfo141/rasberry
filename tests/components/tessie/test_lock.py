"""Test the Tessie lock platform."""

from unittest.mock import patch

import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.lock import (
    DOMAIN as LOCK_DOMAIN,
    SERVICE_LOCK,
    SERVICE_UNLOCK,
)
from homeassistant.const import ATTR_ENTITY_ID, STATE_LOCKED, STATE_UNLOCKED
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ServiceValidationError

from .common import setup_platform


async def test_locks(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
    """Tests that the lock entity is correct."""

    assert len(hass.states.async_all("lock")) == 0

    await setup_platform(hass)

    assert hass.states.async_all("lock") == snapshot(name="all")

    entity_id = "lock.test_lock"

    # Test lock set value functions
    with patch("homeassistant.components.tessie.lock.lock") as mock_run:
        await hass.services.async_call(
            LOCK_DOMAIN,
            SERVICE_LOCK,
            {ATTR_ENTITY_ID: [entity_id]},
            blocking=True,
        )
        mock_run.assert_called_once()
    assert hass.states.get(entity_id).state == STATE_LOCKED

    with patch("homeassistant.components.tessie.lock.unlock") as mock_run:
        await hass.services.async_call(
            LOCK_DOMAIN,
            SERVICE_UNLOCK,
            {ATTR_ENTITY_ID: [entity_id]},
            blocking=True,
        )

        mock_run.assert_called_once()

    # Test charge cable lock set value functions
    entity_id = "lock.test_charge_cable_lock"
    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            LOCK_DOMAIN,
            SERVICE_LOCK,
            {ATTR_ENTITY_ID: [entity_id]},
            blocking=True,
        )

    with patch(
        "homeassistant.components.tessie.lock.open_unlock_charge_port"
    ) as mock_run:
        await hass.services.async_call(
            LOCK_DOMAIN,
            SERVICE_UNLOCK,
            {ATTR_ENTITY_ID: [entity_id]},
            blocking=True,
        )
        assert hass.states.get(entity_id).state == STATE_UNLOCKED
        mock_run.assert_called_once()
