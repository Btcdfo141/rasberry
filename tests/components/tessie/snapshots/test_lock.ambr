# serializer version: 1
# name: test_locks[all]
  list([
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Test Lock',
        'supported_features': <LockEntityFeature: 0>,
      }),
      'context': <ANY>,
      'entity_id': 'lock.test_lock',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'locked',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Test Charge cable lock',
        'supported_features': <LockEntityFeature: 0>,
      }),
      'context': <ANY>,
      'entity_id': 'lock.test_charge_cable_lock',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'locked',
    }),
  ])
# ---
